pkg/transport/manager.go

```diff
@@ -153,9 +176,42 @@ func (tm *Manager) initTransports(ctx context.Context) {
 			tpID   = entry.Entry.ID
 		)
 		if _, err := tm.saveTransport(remote, tpType); err != nil {
-			tm.Logger.Warnf("INIT: failed to init tp: type(%s) remote(%s) tpID(%s)", tpType, remote, tpID)
+			if err != snet.ErrNetworkNotReady {
+				tm.Logger.Warnf("INIT: failed to init tp: type(%s) remote(%s) tpID(%s)", tpType, remote, tpID)
+			}
+
+			// in this case network of `tpType` is not ready which is not permanent, so we're
+			// setting up a goroutine which will try to establish this transport
+			// TODO (nkryuchkov): once transport is generalized we could set up some kind of channel to wait for the network to come up and try to reestablish transport after that. for now we're just retrying to establish transport which is not really efficient
+			tm.Logger.Warnf("INIT: failed to init tp: type(%s) remote(%s) tpID(%s), retrying...", tpType, remote, tpID)
+
+			go func(entry *EntryWithStatus) {
+				retrier := netutil.NewRetrier(tm.Logger, 1*time.Second, 10*time.Second, 0, 1)
+				err := retrier.Do(ctx, func() error {
+					var (
+						tpType = entry.Entry.Type
+						remote = entry.Entry.RemoteEdge(tm.Conf.PubKey)
+					)
+
+					if _, err := tm.saveTransport(remote, tpType); err != nil {
+						if err == snet.ErrNetworkNotReady {
+							// keep on retrying only if the network is not ready.
+							// otherwise transport is either established or considered dead
+							return err
+						}
+					} else {
+						tm.Logger.Infof("Successfully initialized TP %v", *entry.Entry)
+					}
+
+					return nil
+				})
+				if err != nil {
+					tm.Logger.WithError(err).Warnf("INIT: failed to init tp: type(%s) remote(%s) tpID(%s)", tpType, remote, tpID)
+				}
+			}(entry)
+		} else {
+			tm.Logger.Debugf("Successfully initialized TP %v", *entry.Entry)
```

Comment:
If this entire section of code needed? I don't see a problem with the `.Dial` calls returning an error within `managedTransport`.

Reply:
@evanlinjin what if we need to call `Accept`, not `Dial`? we need to know about the transport first

Reply:
at least, I think that this transport initialization is made for that. tbh can't be sure

Reply:
Yes, calls to `.Listen` would fail, but I think that is okay.

Reply:
@evanlinjin I'm sorry I'm a bit lost. I'm starting to wonder why we have this TP initialization logic at all lol. but wouldn't it be crucial to have this retrying logic for intermediary visors? these ones may not have apps running, so they don't really known whether they should call `Dial` or `Accept`, they may only rely on the discovery info. again, correct me if I'm wrong please. I might be getting more and more confused here

Reply:
`saveTransport` adds a `managedTransport` to the `transport.Manager`. A `managedTransport` reconnects itself on failure, so reconnection logic is not needed.

Also, I do not think we need all networks to be working to save transports associated with them. Transports have `UP` and `DOWN` statuses. Let's say `arclient` is not working, then all transports that require `arclient` to function would be `DOWN`.

Reply:
you have your point, probably this would be enough. I'll deal with this part. also please take a look at the query just a bit above @evanlinjin

Reply:
I'm thinking about the above query right now... it's tough haha